src/devices/obs.ts

```diff
@@ -0,0 +1,740 @@
+				// connection error
+				this.emit('error', 'OBS', err)
+				this._setConnected(false)
+				return false
+			})
+	}
+
+	private _connectionChanged () {
+		this.emit('connectionChanged', this.getStatus())
+	}
+
+	private _setConnected (connected: boolean, authenticated: boolean = false) {
+		if (
+			this._connected !== connected ||
+			this._authenticated !== authenticated
+		) {
+			this._connected = connected
+			this._authenticated = authenticated
+			this._connectionChanged()
+
+			if (!this._authenticated) {
+				this._initialized = false
+			}
+		}
+	}
+
+	private _triggerRetryConnection () {
+		if (!this._retryConnectTimeout) {
+			this._retryConnectTimeout = setTimeout(() => {
+				this._retryConnection()
+			}, RETRY_TIMEOUT)
+		}
+	}
+	private _retryConnection () {
+		if (this._retryConnectTimeout) {
+			clearTimeout(this._retryConnectTimeout)
+			this._retryConnectTimeout = undefined
+		}
+
+		if (!this.connected && !this._setDisconnected) {
+			this._connect()
+				.then((connected) => {
+					if (!connected) {
+						this._triggerRetryConnection()
+					}
+				})
+				.catch((err) => {
+					this.emit('error', 'OBS retryConnection', err)
+				})
+		}
+	}
+
+	private _getDefaultState (): OBSState {
+		return {
+			currentScene: undefined,
+			previewScene: undefined,
+			currentTransition: undefined,
+			muted: {},
+			recording: undefined,
+			streaming: undefined,
+			scenes: {},
+			sources: {}
+		}
+	}
+
+	/** Called by the Conductor a bit before a .handleState is called */
+	prepareForHandleState (newStateTime: number) {
+		// clear any queued commands later than this time:
+		this._doOnTime.clearQueueNowAndAfter(newStateTime + 0.1)
+		this.cleanUpStates(0, newStateTime + 0.1)
+	}
+
+	handleState (newState: TimelineState, newMappings: Mappings) {
+		super.onHandleState(newState, newMappings)
+		if (!this._initialized) {
+			// before it's initialized don't do anything
+			this.emit('warning', 'OBS not initialized yet')
+			return
+		}
+
+		let previousStateTime = Math.max(
+			this.getCurrentTime() + 0.1,
+			newState.time
+		)
+		let oldState: OBSState = (
+			this.getStateBefore(previousStateTime) || {
+				state: this._getDefaultState()
+			}
+		).state
+
+		let newOBSState = this.convertStateToOBS(newState, newMappings)
+
+		let commandsToAchieveState: Array<any> = this._diffStates(
+			oldState,
+			newOBSState
+		)
+
+		// clear any queued commands later than this time:
+		this._doOnTime.clearQueueNowAndAfter(previousStateTime)
+
+		// add the new commands to the queue:
+		this._addToQueue(commandsToAchieveState, newState.time)
+
+		// store the new state, for later use:
+		this.setState(newOBSState, newState.time)
+	}
+
+	clearFuture (clearAfterTime: number) {
+		// Clear any scheduled commands after this time
+		this._doOnTime.clearQueueAfter(clearAfterTime)
+	}
+
+	async terminate () {
+		this._setDisconnected = true
+		this._doOnTime.dispose()
+		this._obs.disconnect()
+		return Promise.resolve(true)
```

Comment:
As you're inside an async function, this could be simple `return true`. Up to you :)